course5/homework11.1/homework11.1.cc

```diff
@@ -0,0 +1,64 @@
+#include <algorithm>
+#include <vector>
+#include <list>
+#include <deque>
+#include <benchmark/benchmark.h>
+
+#include "linked_list.h"
+
+template <typename T>
+static void push_back(T & container, int element)
+{
+	container.push_back(element);
+}
+
+static void push_back(jds::LinkedList<int> & list, int element)
+{
+	std::vector<int> temp;
+	while (list.size())
+	{
+		temp.push_back(list.front());
+		list.pop_front();
+	}
+	list.push_front(element);
+	while (!temp.empty())
+	{
+		list.push_front(temp.back());
+		temp.pop_back();
+	}
```

Comment:
You could just ```push_front``` here. The order in which the integers sit in the container doesn't matter for the outcome. Or does it?